core/src/snapshot_packager_service/snapshot_gossip_manager.rs

```diff
@@ -26,120 +24,153 @@ impl SnapshotGossipManager {
-            self.incremental_snapshot_hashes.base = incremental_snapshot_hash.base;
-        }
+        self.update_latest_incremental_snapshot_hash(incremental_snapshot_hash, base_slot);
+        self.push_latest_snapshot_hashes_to_cluster();
+    }
 
-        self.incremental_snapshot_hashes
-            .hashes
-            .push(incremental_snapshot_hash.hash);
+    /// Update the latest snapshot hashes with a new full snapshot
+    fn update_latest_full_snapshot_hash(&mut self, full_snapshot_hash: (Slot, SnapshotHash)) {
+        self.latest_snapshot_hashes = Some(LatestSnapshotHashes {
+            full: full_snapshot_hash,
+            incremental: None,
+        });
+    }
 
-        retain_max_n_elements(
-            &mut self.incremental_snapshot_hashes.hashes,
-            self.max_incremental_snapshot_hashes,
+    /// Update the latest snapshot hashes with a new incremental snapshot
+    fn update_latest_incremental_snapshot_hash(
+        &mut self,
+        incremental_snapshot_hash: (Slot, SnapshotHash),
+        base_slot: Slot,
+    ) {
+        let latest_snapshot_hashes = self
+            .latest_snapshot_hashes
+            .as_mut()
+            .expect("there must already be a full snapshot hash");
+        assert_eq!(
+            base_slot, latest_snapshot_hashes.full.0,
+            "the incremental snapshot's base slot ({}) must match the latest full snapshot's slot ({})",
+            base_slot, latest_snapshot_hashes.full.0,
         );
+        latest_snapshot_hashes.incremental = Some(incremental_snapshot_hash);
+    }
 
-        // Pushing incremental snapshot hashes to the cluster should never fail.  The only error
-        // case is when the length of the hashes is too big, but we account for that with
-        // `max_incremental_snapshot_hashes`.  If this call ever does error, it's a programmer bug!
-        // Check to see what changed in `push_snapshot_hashes()` and handle the new
-        // error condition here.
+    /// Push the latest snapshot hashes to the cluster via CRDS
+    fn push_latest_snapshot_hashes_to_cluster(&self) {
+        let Some(latest_snapshot_hashes) = self.latest_snapshot_hashes.as_ref() else {
+            return;
+        };
+
+        // Pushing snapshot hashes to the cluster should never fail.  The only error case is when
+        // the length of the incremental hashes is too big, (and we send a maximum of one here).
+        // If this call ever does error, it's a programmer bug!  Check to see what changed in
+        // `push_snapshot_hashes()` and handle the new error condition here.
         self.cluster_info
             .push_snapshot_hashes(
-                clone_hash_for_crds(&self.incremental_snapshot_hashes.base),
-                clone_hashes_for_crds(&self.incremental_snapshot_hashes.hashes),
+                clone_hash_for_crds(&latest_snapshot_hashes.full),
+                latest_snapshot_hashes
+                    .incremental
```

Comment:
Is this guaranteed to be either `None` or a single incremental snapshot hash? Seems like we could do something besides iter/map/collect. Future cleanup?

Reply:
And we're keeping it a `Vec` to be compatible with existing Gossip stack / CRDS types, right? Any plans to trim this down in the future?